projects/etos_suite_runner/src/etos_suite_runner/lib/suite.py

```diff
@@ -152,14 +155,17 @@ def start(self, identifier: str, otel_context_carrier: dict) -> None:
         finally:
             opentelemetry.context.detach(otel_context_token)
 
-    def release(self, testrun_id) -> None:
-        """Release this sub suite."""
+    def _delete_environment(self) -> bool:
+        """Delete environment from Kubernetes."""
+        environment_name = self.environment.get("executor", {}).get("id")
+        environment_client = Environment(Kubernetes(), strict=True)
+        return environment_client.delete(environment_name)
```

Comment:
This will involve checkins? Do we have any safety mechanisms here? Retries or similar?

Reply:
Retries and safety mechanisms are implemented in the EnvironmentController